src/components/FilterBar.jsx

```diff
@@ -30,38 +30,40 @@ const grades = {
   16: 'College Senior',
 };
 
-const sortingOptions = {
-  deadlineSoon: 'Deadline (Earliest to Latest)',
-  deadlineLatest: 'Deadline (Latest to Earliest)',
-  amountLow: 'Amount (Low to High)',
-  amountHigh: 'Amount (High to Low)',
+const sortItems = {
+  'deadline.asc': 'Deadline (Earliest to Latest)',
+  'deadline.desc': 'Deadline (Latest to Earliest)',
+  'amount.asc': 'Amount (Low to High)',
+  'amount.desc': 'Amount (High to Low)',
+};
+
+const sortOptions = {
+  'deadline.asc': {
+    field: 'deadline',
+    dir: 'asc',
+  },
+  'deadline.desc': {
+    field: 'deadline',
+    dir: 'desc',
+  },
+  'amount.asc': {
+    field: 'amount.min',
+    dir: 'asc',
+  },
+  'amount.desc': {
+    field: 'amount.max',
+    dir: 'desc',
+  },
 };
 
 export default function FilterBar(props) {
-  const { changeSortBy, changeSortFormat, setQueryParam, queryParams } = props;
+  const { setQueryParam, queryParams } = props;
   const classes = useStyles();
-
   const { minAmount, maxAmount } = queryParams;
 
-  function updateSorting(sortingOption) {
-    switch (sortingOption) {
-      case 'deadlineSoon':
-        changeSortBy('deadline');
-        changeSortFormat('asc');
-        break;
-      case 'deadlineLatest':
-        changeSortBy('deadline');
-        changeSortFormat('desc');
-        break;
-      case 'amountLow':
-        changeSortBy('amount.min');
-        changeSortFormat('asc');
-        break;
-      case 'amountHigh':
-        changeSortBy('amount.max');
-        changeSortFormat('desc');
-        break;
-      default:
+  function updateSortingParams(sortOption) {
+    if (sortOption in sortOptions) {
```

Comment:
Per comment above, this check would no longer be needed if those changes are made.